fenix/app/src/main/java/org/mozilla/fenix/AppNavigationBinding.kt

```diff
@@ -0,0 +1,122 @@
+import org.mozilla.fenix.addons.AddonPermissionsDetailsFragmentDirections
+import org.mozilla.fenix.addons.AddonsManagementFragmentDirections
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppState
+import org.mozilla.fenix.components.appstate.Screen
+import org.mozilla.fenix.exceptions.trackingprotection.TrackingProtectionExceptionsFragmentDirections
+import org.mozilla.fenix.ext.alreadyOnDestination
+import org.mozilla.fenix.ext.nav
+import org.mozilla.fenix.home.HomeFragment
+import org.mozilla.fenix.home.HomeFragmentDirections
+import org.mozilla.fenix.library.bookmarks.BookmarkFragmentDirections
+import org.mozilla.fenix.library.history.HistoryFragmentDirections
+import org.mozilla.fenix.library.historymetadata.HistoryMetadataGroupFragmentDirections
+import org.mozilla.fenix.library.recentlyclosed.RecentlyClosedFragmentDirections
+import org.mozilla.fenix.search.SearchDialogFragmentDirections
+import org.mozilla.fenix.settings.CookieBannersFragmentDirections
+import org.mozilla.fenix.settings.HttpsOnlyFragmentDirections
+import org.mozilla.fenix.settings.SettingsFragmentDirections
+import org.mozilla.fenix.settings.TrackingProtectionFragmentDirections
+import org.mozilla.fenix.settings.about.AboutFragmentDirections
+import org.mozilla.fenix.settings.logins.fragment.LoginDetailFragmentDirections
+import org.mozilla.fenix.settings.logins.fragment.SavedLoginsAuthFragmentDirections
+import org.mozilla.fenix.settings.search.SaveSearchEngineFragmentDirections
+import org.mozilla.fenix.settings.search.SearchEngineFragmentDirections
+import org.mozilla.fenix.settings.studies.StudiesFragmentDirections
+import org.mozilla.fenix.settings.wallpaper.WallpaperSettingsFragmentDirections
+import org.mozilla.fenix.share.AddNewDeviceFragmentDirections
+import org.mozilla.fenix.shopping.ReviewQualityCheckFragmentDirections
+import org.mozilla.fenix.tabstray.TabsTrayFragmentDirections
+import org.mozilla.fenix.trackingprotection.TrackingProtectionPanelDialogFragmentDirections
+
+class AppNavigationBinding(
+    appStore: AppStore,
+    private val navController: NavController,
+) : AbstractBinding<AppState>(appStore) {
+    override suspend fun onState(flow: Flow<AppState>) = flow.distinctUntilChangedBy { it.screen }
+        .collect {
+            when (val screen = it.screen) {
+                is Screen.Home -> {
+                    navController.navigate(R.id.homeFragment)
+                }
+                is Screen.Browser -> {
+                    val from = screen.from
+                    if (navController.alreadyOnDestination(R.id.browserFragment)) return@collect
+                    @IdRes val fragmentId = if (from.fragmentId != 0) from.fragmentId else null
+                    val directions = getNavDirections(from, screen.customTabSessionId)
+                    navController.nav(fragmentId, directions)
+                }
+            }
+        }
+
+    private fun getNavDirections(
```

Comment:
I'm not actually sure of the benefit of these `NavDirections`. All the ones that I checked seemed to reference the area they were called from, which would mean we could just push new Destinations onto the stack and rely on normal popping behavior.

All that to say, I think there's a possibility that all of this could be removed to reduce the code in this file.